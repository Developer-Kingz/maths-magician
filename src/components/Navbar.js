import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => (
  <div>
    <div className="navbar">
      <div className="logo">
        <h1>Math Magician</h1>
      </div>
      <div className="nav">
        <ul>
          <NavLink to="/">
            <li>Home</li>
          </NavLink>
          <NavLink to="/calculator">
            <li>Calculator</li>
          </NavLink>
          <NavLink to="/quotes">
            <li>Quotes</li>
          </NavLink>
        </ul>
      </div>
    </div>
  </div>
);

export default Navbar;
