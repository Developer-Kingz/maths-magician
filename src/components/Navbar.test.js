import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter to wrap the Navbar component
import '@testing-library/jest-dom/extend-expect'; // For extending expect with additional matchers
import Navbar from './Navbar';

test('renders Navbar correctly', () => {
  // Arrange (Render the Navbar component within the MemoryRouter)
  const { asFragment } = render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>,
  );

  // Act (No action needed for this test)

  // Assert (Check if the rendered output matches the snapshot)
  expect(asFragment()).toMatchSnapshot();
});
