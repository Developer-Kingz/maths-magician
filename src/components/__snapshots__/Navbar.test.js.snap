// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="navbar"
    >
      <div
        class="logo"
      >
        <h1>
          Math Magician
        </h1>
      </div>
      <div
        class="nav"
      >
        <ul>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <li>
              Home
            </li>
          </a>
          <a
            class=""
            href="/calculator"
          >
            <li>
              Calculator
            </li>
          </a>
          <a
            class=""
            href="/quotes"
          >
            <li>
              Quotes
            </li>
          </a>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
